### PR TITLE
Feature/5 expense api

### PR DESCRIPTION
**Pull request by Mouiad-JRA**

### CODE REVIEW

File: backend/expensia/expense/admin.py [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
+    list_display = ('name', 'amount', 'start_date', 'end_date',)
+    list_filter = ('start_date', 'end_date', 'category')
+    filter_horizontal = ('category',)
+    order = ['-start_date']
+    fieldsets = (
+        (None, {'fields': (
+            'name', 'amount', 'start_date', 'end_date', 'category',)}),
+    )
+
+    def has_add_permission(self, request, obj=None):
```
Comment (HHHMHA) [inline backend/expensia/expense/admin.py:21]: not needed

File: backend/expensia/expense/admin.py [hunk 2]
```diff
@@ -1,5 +1,29 @@
...
+        return True
+
+    def get_list_display(self, request):
```
Comment (HHHMHA) [inline backend/expensia/expense/admin.py:24]: not needed

File: backend/expensia/expense/models.py [hunk 3]
```diff
@@ -11,3 +14,30 @@ def __str__(self):
 
     def get_absolute_url(self):
         return reverse('expense:category-detail', args=(self.pk,))
+
+
+class Expense(models.Model):
+    """ Expense model """
+
+    name =models.CharField(_("Name"), max_length=30, blank=False, null=False)
+    start_date = models.DateTimeField(_("Start Date"), null=False, blank=False)
```
Comment (HHHMHA) [inline backend/expensia/expense/models.py:23]: Expense just has a creation date no need for start and end date here.
Just inherit from `TimeStampedModel`

Don't forget to remove these two fields in other places.
Also delete the migration and create another one

File: backend/expensia/expense/models.py [hunk 4]
```diff
@@ -11,3 +14,30 @@ def __str__(self):
...
+    def __str__(self):
+        return self.name
+
+    def get_absolute_url(self):
+        return reverse("expense:expense_details", args=(self.id,))
+
+    def get_amount(self):
+        return Decimal(self.amount)
+
+    def get_categories(self):
```
Comment (HHHMHA) [inline backend/expensia/expense/models.py:39]: Each expense has one category so what does this method do?

File: backend/expensia/expense/views.py [hunk 5]
```diff
@@ -9,3 +9,16 @@ class ExpenseCategoryView(ModelViewSet):
...
+
+
+class ExpenseViewSet(ModelViewSet):
+    queryset = Expense.objects.all()
+    serializer_class = ExpenseSerializer
+    permission_classes = (AllowAny,)
+    filter_fields = ("start_date", "end_date")
+    lookup_field = "id"
+
+    def get_serializer_context(self):
```
Comment (HHHMHA) [inline backend/expensia/expense/views.py:21]: Not needed

File: backend/expensia/expense/views.py [hunk 6]
```diff
@@ -9,3 +9,16 @@ class ExpenseCategoryView(ModelViewSet):
     serializer_class = CategorySerializer
     permission_classes = (AllowAny,)
     authentication_classes = ()
+
+
+class ExpenseViewSet(ModelViewSet):
+    queryset = Expense.objects.all()
+    serializer_class = ExpenseSerializer
+    permission_classes = (AllowAny,)
+    filter_fields = ("start_date", "end_date")
```
Comment (HHHMHA) [inline backend/expensia/expense/views.py:18]: These will become custom filtering
you can create a FilterClass for them or do it manually in the `filter_queryset `method

File: backend/requirements/base.txt [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# Django
```
Comment (HHHMHA) [inline backend/requirements/base.txt:1]: Not needed also not updated

**Review by HHHMHA — CHANGES_REQUESTED** [review]

**Comment by Mouiad-JRA** [comment]
Don't fuck with me , Fix the plugin error  and update the req

**Review by HHHMHA — APPROVED** [review]

**Pull request merged**